From: frode.nordahl@canonical.com
Subject: Run a system test using already installed components

The purpose of this patch is to make the first system test run
using already installed components for ovs-vswitchd (and its db)
and ovn-controller.

We want this to be able to confirm operation of OVS and OVN when
running confined within the ovn-chassis snap.

diff --git a/tests/ovs-macros.at b/tests/ovs-macros.at
index 0482b7f5b..c95bad5de 100644
--- a/tests/ovs-macros.at
+++ b/tests/ovs-macros.at
@@ -1,4 +1,3 @@
-AT_TESTED([ovs-vswitchd])
 AT_TESTED([ovs-vsctl])
 
 m4_include([m4/compat.m4])
diff --git a/tests/system-ovn.at b/tests/system-ovn.at
index 3d2591ee9..2ba169690 100644
--- a/tests/system-ovn.at
+++ b/tests/system-ovn.at
@@ -1,13 +1,11 @@
 AT_BANNER([system-ovn])
 
-OVN_FOR_EACH_NORTHD([
 AT_SETUP([2 LRs connected via LS, gateway router, SNAT and DNAT])
 AT_KEYWORDS([ovnnat])
 
 CHECK_CONNTRACK()
 CHECK_CONNTRACK_NAT()
 ovn_start
-OVS_TRAFFIC_VSWITCHD_START()
 ADD_BR([br-int])
 
 # Set external-ids in br-int needed for ovn-controller
@@ -18,9 +16,6 @@ ovs-vsctl \
         -- set Open_vSwitch . external-ids:ovn-encap-ip=169.0.0.1 \
         -- set bridge br-int fail-mode=secure other-config:disable-in-band=true
 
-# Start ovn-controller
-start_daemon ovn-controller
-
 # Logical network:
 # Two LRs - R1 and R2 that are connected to each other via LS "join"
 # in 20.0.0.0/24 network. R1 has switches foo (192.168.1.0/24) and
@@ -161,23 +156,7 @@ AT_CHECK([ovs-appctl dpctl/dump-conntrack | FORMAT_CT(30.0.0.1) | \
 sed -e 's/zone=[[0-9]]*/zone=<cleared>/'], [0], [dnl
 icmp,orig=(src=192.168.2.2,dst=192.168.1.2,id=<cleared>,type=8,code=0),reply=(src=192.168.1.2,dst=30.0.0.1,id=<cleared>,type=0,code=0),zone=<cleared>
 ])
-
-OVS_APP_EXIT_AND_WAIT([ovn-controller])
-
-as ovn-sb
-OVS_APP_EXIT_AND_WAIT([ovsdb-server])
-
-as ovn-nb
-OVS_APP_EXIT_AND_WAIT([ovsdb-server])
-
-as northd
-OVS_APP_EXIT_AND_WAIT([NORTHD_TYPE])
-
-as
-OVS_TRAFFIC_VSWITCHD_STOP(["/failed to query port patch-.*/d
-/connection dropped.*/d"])
 AT_CLEANUP
-])
 
 OVN_FOR_EACH_NORTHD([
 AT_SETUP([2 LRs connected via LS, gateway router, SNAT and DNAT - IPv6])
